torchdp/privacy_engine.py

```diff
@@ -95,16 +111,36 @@ def get_privacy_spent(self, target_delta: float = None):
 
     def step(self):
         self.steps += 1
-        clip_values = self.clipper.step()
+        clip_values, batch_size = self.clipper.step()
+
+        # ensure the clipper consumed the right amount of gradients.
+        # In the last batch of a training epoch, we might get a batch that is
+        # smaller than others but we should never get a batch that is too large
+        if batch_size > self.batch_size:
+            raise ValueError(
+                f"PrivacyEngine expected a batch of size {self.batch_size} "
+                f"but received a batch of size {batch_size}"
+            )
+
+        if batch_size < self.batch_size:
+            warnings.warn(
+                f"PrivacyEngine expected a batch of size {self.batch_size} "
+                f"but received a batch of size {batch_size}. The privacy "
```

Comment:
... But the last step had batch_size = {batch_size}. This means we will be a bit more pessimistic in our analysis. Consider setting `drop_last = True` in your PyTorch dataloader to avoid this problem completely

Reply:
Done. Thanks